File: user/src/main/kotlin/kpring/user/service/LoginService.java
package kpring.user.service;

import kpring.user.dto.request.LoginRequest;
import kpring.user.dto.request.LogoutRequest;
import kpring.user.dto.result.LoginResponse;
import org.springframework.stereotype.Service;

@Service
public class LoginService {

    /**
     * @throws IllegalArgumentException if request is invalid
     * @throws RuntimeException         if an unexpected error occurs
     */
    public LoginResponse login(LoginRequest request) {
        return null;
    }

    /**
     * @throws IllegalArgumentException if request is invalid
     * @throws RuntimeException         if an unexpected error occurs
     */
    public void logout(LogoutRequest request) {
    }
}

File: user/src/main/kotlin/kpring/user/service/LoginService.kt
package kpring.user.service

import kpring.core.user.dto.request.LoginRequest
import kpring.core.user.dto.request.LogoutRequest
import kpring.core.user.dto.response.LoginResponse

interface LoginService {
  fun login(request: LoginRequest): LoginResponse

  fun logout(request: LogoutRequest)
}
